u8timeseries/timeseries.py

```diff
@@ -38,26 +38,33 @@ def __init__(self,
             Optionally, a Pandas Series representing lower confidence interval.
         confidence_hi
             Optionally, a Pandas Series representing upper confidence interval.
+        freq
+            Optionally, a string representing the frequency of the Pandas Series. When creating a TimeSeries
+            instance with a length smaller than 3, this argument must be passed.
         fill_missing_dates
             Optionally, a boolean indicating filling missing dates with NaN in case missing inferred_freq on index.
         """
 
-        # cannot create a timeseries with n<3 -> can add less than 3 elements with add function
-        raise_if_not(len(series) >= 3, 'Series must have at least three values.', logger)
+        raise_if_not(len(series) > 0, 'Series must not be empty.', logger)
         raise_if_not(isinstance(series.index, pd.DatetimeIndex), 'Series must be indexed with a DatetimeIndex.', logger)
         raise_if_not(np.issubdtype(series.dtype, np.number), 'Series must contain numerical values.', logger)
+        raise_if_not(len(series) >= 3 or freq is not None, 'Series must have at least 3 values if the "freq" argument'
+                     'is not passed', logger)
 
-        series = series.sort_index()  # Sort by time
-        if not series.index.inferred_freq:
-            if fill_missing_dates:
-                series = self._fill_missing_dates(series)
-            else:
-                raise_if_not(False, 'Could not infer frequency. Are some dates missing? '
-                                    'Is Series too short (n=2)?', logger)
+        self._series = series.sort_index()  # Sort by time
 
-        self._series: pd.Series = series
-
-        self._freq: str = self._series.index.inferred_freq  # Infer frequency
+        if (len(series) < 3):
+            self._freq: str = freq
+        else:
+            if not series.index.inferred_freq:
+                if fill_missing_dates:
+                    self._series = self._fill_missing_dates(self._series)
+                else:
+                    raise_if_not(False, 'Could not infer frequency. Are some dates missing? '
+                                        'Is Series too short (n=2)?', logger)
```

Comment:
You can remove the `Is Series too short` part from this message. You can maybe tell the user something like:
```
Could not infer frequency. Are some dates missing? Try specifying a `freq` when building the `TimeSeries`, or specify `fill_missing_dates=True`.
```

Reply:
How about just `Could not infer frequency. Are some dates missing? Try specifying `fill_missing_dates=True`.`? Because the way it is implemented right now, if a series has a length of at least 3, the frequency will always be set by calling infer_freq.